File: plugins/hogyeong-kakao-map/ios/HogyeongKakaoMapViewManager.m
#import <React/RCTViewManager.h>
#import "hogyeongcrew-Swift.h"

@interface HogyeongKakaoMapViewManager : RCTViewManager
@end

@implementation HogyeongKakaoMapViewManager

RCT_EXPORT_MODULE(HogyeongKakaoMapView)

+ (BOOL)requiresMainQueueSetup
{
  return YES;
}

- (UIView *)view
{
  return [HogyeongKakaoMapView new];
}

RCT_EXPORT_VIEW_PROPERTY(appKey, NSString)
RCT_EXPORT_VIEW_PROPERTY(camera, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(currentLocation, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(routeCoordinates, NSArray)
RCT_EXPORT_VIEW_PROPERTY(photoMarkers, NSArray)

@end

File: plugins/hogyeong-kakao-map/android/HogyeongKakaoMapViewManager.kt
package com.quigmire1.hogyeongcrew.map

import com.facebook.react.bridge.ReadableArray
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp

class HogyeongKakaoMapViewManager : SimpleViewManager<HogyeongKakaoMapView>() {
  override fun getName(): String = REACT_CLASS

  override fun createViewInstance(reactContext: ThemedReactContext): HogyeongKakaoMapView =
    HogyeongKakaoMapView(reactContext)

  @ReactProp(name = "appKey")
  fun setAppKey(view: HogyeongKakaoMapView, value: String?) {
    view.setAppKey(value)
  }

  @ReactProp(name = "camera")
  fun setCamera(view: HogyeongKakaoMapView, value: ReadableMap?) {
    view.setCamera(value)
  }

  @ReactProp(name = "currentLocation")
  fun setCurrentLocation(view: HogyeongKakaoMapView, value: ReadableMap?) {
    view.setCurrentLocation(value)
  }

  @ReactProp(name = "routeCoordinates")
  fun setRouteCoordinates(view: HogyeongKakaoMapView, value: ReadableArray?) {
    view.setRouteCoordinates(value)
  }

  @ReactProp(name = "photoMarkers")
  fun setPhotoMarkers(view: HogyeongKakaoMapView, value: ReadableArray?) {
    view.setPhotoMarkers(value)
  }

  companion object {
    const val REACT_CLASS = "HogyeongKakaoMapView"
  }
}
